src/kinetics/Reaction.cpp

```diff
@@ -196,9 +196,11 @@ void Reaction::check()
 AnyMap Reaction::parameters(bool withInput) const
 {
     AnyMap out;
+    static set<string> excludes{"type", "duplicate", "orders", "negative-orders",
+                                "nonreactant-orders"};
     getParameters(out);
     if (withInput) {
-        out.update(input);
+        out.update(input, true, excludes);
```

Comment:
While adding `excludes` resolves the issue at hand, I am not convinced that it is the most robust implementation - you are essentially specifying what must not be overwritten, which needs to be adapted for any future change of parameters.

I honestly would prefer the following alternative: if `withInput` is set, initialize `out` with contents of `input` (empty otherwise) and _then_ overwrite things with `getParameter`. While it may take more effort (I actually tried that a couple of weeks ago, and ran into a buzz saw of broken logic), it is imho the cleaner resolution, as it will always overwrite things the correct way, irrespective of future changes.

Reply:
Yeah, I tried that approach as well. I was able to fix the logic in terms of _what_ gets output, but it really messed up control over the order to output the keys. Currently, our logic is to output programmatically added keys first, followed by user-defined keys. However, if we start by inserting all of the raw data, then `AnyMap` sees all of the keys that existed in `input` as "user-defined", and there's not really a way to change their state, since later replacing a value isn't easily distinguishable from just accessing it, since both are just invocations of `AnyMap::operator[]`.

One other option I considered was to delete these defaultable elements out of `input` as we read them, but that changes the interpretation of what `input` is, and causes problems in a couple cases where we later want to access the original item to show it as context for an error message.

I agree that it would be preferable to keep any specification of keys to be excluded with the code in `setParameters`/`getParameters`. I _think_ I have another idea. Stay tuned...